src/org/skypro/skyshop/basket/ProductBasket.java

```diff
@@ -78,28 +78,42 @@ public void searchProductInBasket(Product product) {
     }
 
     public void cleanBasket() {
-        Iterator<Product> itrBasket = basket.iterator();
         if (basket.isEmpty()) {
             System.out.println("В корзине пусто");
         } else {
-            while (itrBasket.hasNext()) {
-                itrBasket.next();
-                itrBasket.remove();
-            }
+            basket.clear();
             System.out.println("Корзина очищена");
         }
     }
 
-    public void removeProductFromBasket(String name) {
+    public List<Product> removeProductFromBasket(String name) {
+        List<Product> listRemove = new LinkedList<>();
         Iterator<Product> itrBasket = basket.iterator();
         if (basket.isEmpty()) {
-            System.out.println("В корзине пусто");
+            listRemove = null;
```

Comment:
по заданию надо пустой список вернуть, а не null
```suggestion
            listRemove = Collections.emptyList();
```